tests/tasks/hightouch/test_hightouch_tasks.py

```diff
@@ -0,0 +1,136 @@
+        with pytest.raises(ValueError, match=msg_match):
+            ht_task.run()
+
+    def test_run_missing_api_key_with_api_key_env_var(self):
+        ht_task = HightouchRunSync()
+        msg_match = "`api_key` is missing and `api_key_env_var` cannot be found."
+        with pytest.raises(ValueError, match=msg_match):
+            ht_task.run(api_key_env_var="foo")
+
+    def test_run_missing_sync_id(self):
+        ht_task = HightouchRunSync()
+        msg_match = "`sync_id` is missing."
+        with pytest.raises(ValueError, match=msg_match):
+            ht_task.run(api_key="key")
+
+    @responses.activate
+    def test_run_with_api_key(self):
+        ht_task = HightouchRunSync()
+        sync_id = 123
+        responses.add(
+            responses.POST,
+            f"https://api.hightouch.io/api/v2/rest/run/{sync_id}",
+            status=200,
+            json={"message": "it works"},
+        )
+
+        response = ht_task.run(api_key="key", sync_id=sync_id)
+
+        assert responses.calls[0].request.headers["Authorization"] == "Bearer key"
+        assert isinstance(response, dict)
+
+    @patch.dict(os.environ, {"foo": "bar"})
+    @responses.activate
+    def test_run_with_api_key_env_var(self):
+        ht_task = HightouchRunSync()
+        sync_id = 123
+        responses.add(
+            responses.POST,
+            f"https://api.hightouch.io/api/v2/rest/run/{sync_id}",
+            status=200,
+            json={"message": "it works"},
+        )
+
+        response = ht_task.run(api_key="key", sync_id=sync_id, api_key_env_var="foo")
+
+        assert responses.calls[0].request.headers["Authorization"] == "Bearer key"
+        assert isinstance(response, dict)
+
+    @responses.activate
+    def test_run_raises_on_status_code_not_ok(self):
+        ht_task = HightouchRunSync()
+        sync_id = 123
+        responses.add(
+            responses.POST,
+            f"https://api.hightouch.io/api/v2/rest/run/{sync_id}",
+            status=123,
+        )
+
+        msg = "Error while starting sync run."
+        with pytest.raises(FAIL, match=msg):
+            ht_task.run(api_key="key", sync_id=sync_id)
+
+    @responses.activate
+    def test_run_with_wait(self):
+        ht_task = HightouchRunSync()
+        sync_id = 123
+        responses.add(
+            responses.POST,
+            f"https://api.hightouch.io/api/v2/rest/run/{sync_id}",
+            status=200,
+            json={"message": "it works"},
+        )
+        responses.add(
+            responses.GET,
+            f"https://api.hightouch.io/api/v2/rest/sync/{sync_id}",
+            status=200,
+            json={"sync": {"sync_status": "success"}},
+        )
+
+        response = ht_task.run(api_key="key", sync_id=123, wait_for_completion=True)
+
+        assert (
+            responses.assert_call_count(
+                f"https://api.hightouch.io/api/v2/rest/run/{sync_id}", 1
+            )
+            is True
+        )
```

Comment:
This should raise an `AssertionError` if the call count doesn't match.
```suggestion
            responses.assert_call_count(f"https://api.hightouch.io/api/v2/rest/run/{sync_id}", 1)
```